code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+			itemcolor = "#2956B2"
+		if("pyrite")
+			itemcolor = "#FAFAD2"
+		if("red")
+			itemcolor = "#FF0000"
+		if("green")
+			itemcolor = "#00FF00"
+		if("pink")
+			itemcolor = "#FF69B4"
+		if("gold")
+			itemcolor = "#FFD700"
+		if("oil")
+			itemcolor = "#505050"
+		if("black")
+			itemcolor = "#000000"
+		if("light pink")
+			itemcolor = "#FFB6C1"
+		if("adamantine")
+			itemcolor = "#008B8B"
+	add_atom_colour(itemcolor, FIXED_COLOUR_PRIORITY)
+	START_PROCESSING(SSobj,src)
+
+/obj/structure/slime_crystal/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	for(var/X in affected_mobs)
+		on_mob_leave(X)
+	return ..()
+
+/obj/structure/slime_crystal/process()
+	for(var/mob/living/mob_in_range in view_or_range(3,src,range_type))
+		if(!(mob_in_range in affected_mobs))
+			on_mob_enter(mob_in_range)
+			affected_mobs[mob_in_range] = 0
+
+		affected_mobs[mob_in_range]++
+		on_mob_effect(mob_in_range)
+
+	for(var/M in affected_mobs)
+		if(get_dist(M,src) <= 5)
+			continue
+		on_mob_leave(M)
+		affected_mobs -= M
+
+/obj/structure/slime_crystal/proc/master_crystal_destruction()
+	qdel(src)
+
+/obj/structure/slime_crystal/proc/on_mob_enter(mob/living/affected_mob)
+	return
+
+/obj/structure/slime_crystal/proc/on_mob_effect(mob/living/affected_mob)
+	return
+
+/obj/structure/slime_crystal/proc/on_mob_leave(mob/living/affected_mob)
+	return
+
+/obj/structure/slime_crystal/grey
+	colour = "grey"
+	range_type = "view"
+
+/obj/structure/slime_crystal/grey/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!istype(affected_mob,/mob/living/simple_animal/slime))
+		return
+	var/mob/living/simple_animal/slime/slime_mob = affected_mob
+	slime_mob.nutrition += 2
+
+/obj/structure/slime_crystal/orange
+	colour = "orange"
+	range_type = "view"
+
+/obj/structure/slime_crystal/orange/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!istype(affected_mob,/mob/living/carbon))
+		return
+	var/mob/living/carbon/carbon_mob = affected_mob
+	carbon_mob.fire_stacks++
+	carbon_mob.IgniteMob()
+
+/obj/structure/slime_crystal/orange/process()
+	. = ..()
+	var/turf/open/T = get_turf(src)
+	if(!istype(T))
+		return
+	var/datum/gas_mixture/gas = T.return_air()
+	gas.set_temperature(T0C + 200)
+	T.air_update_turf()
+
+/obj/structure/slime_crystal/purple
+	colour = "purple"
+
+	var/heal_amt = 2
+
+/obj/structure/slime_crystal/purple/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!istype(affected_mob,/mob/living/carbon))
+		return
+	var/mob/living/carbon/carbon_mob = affected_mob
+	var/rand_dam_type = rand(0,10)
+
+	switch(rand_dam_type)
+		if(0)
+			carbon_mob.adjustBruteLoss(-heal_amt)
+		if(1)
+			carbon_mob.adjustFireLoss(-heal_amt)
+		if(2)
+			carbon_mob.adjustOxyLoss(-heal_amt)
+		if(3)
+			carbon_mob.adjustToxLoss(-heal_amt)
+		if(4)
+			carbon_mob.adjustCloneLoss(-heal_amt)
+		if(5)
+			carbon_mob.adjustStaminaLoss(-heal_amt)
+		if(6 to 10)
+			carbon_mob.adjustOrganLoss(pick(ORGAN_SLOT_BRAIN,ORGAN_SLOT_HEART,ORGAN_SLOT_LIVER,ORGAN_SLOT_LUNGS),-heal_amt)
```

Comment:
Would help with visibility if you created the healing particle effects around people being healed by it.